src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509CertificateCollection.cs

```diff
@@ -68,33 +52,26 @@ bool IList.IsReadOnly
             get { return false; }
         }
 
-        Object IList.this[int index]
+        object IList.this[int index]
         {
             get
             {
-                if (index < 0 || index >= Count)
-                    throw new ArgumentOutOfRangeException("index", SR.ArgumentOutOfRange_Index);
-
                 return _list[index];
             }
             set
             {
-                if (index < 0 || index >= Count)
-                    throw new ArgumentOutOfRangeException("index", SR.ArgumentOutOfRange_Index);
                 if (value == null)
                     throw new ArgumentNullException("value");
 
-                _list[index] = value;
+                ((IList)_list)[index] = value;
```

Comment:
Or, I guess the answer is an easy "Because this way it compiles"; and it's only a problem of reading the diff.

I wonder why the indexer is up here when the rest of the IList explicit implementations are later... because when you see lots of the casting it makes more sense :).